### PR TITLE
Refine HV function name

### PR DESCRIPTION
**Pull request by liuyuan-GitHub**
Refine functions name and remove pci_vdev_read_cfg_u8/16/32

Tracked-On: #4433

Signed-off-by: Yuan Liu <yuan1.liu@intel.com>
Acked-by: Eddie Dong <eddie.dong@intel.com>

### CODE REVIEW

File: hypervisor/dm/vpci/vpci.c [hunk 1]
```diff
@@ -39,15 +39,15 @@
 static void vpci_init_vdevs(struct acrn_vm *vm);
 static void deinit_prelaunched_vm_vpci(struct acrn_vm *vm);
 static void deinit_postlaunched_vm_vpci(struct acrn_vm *vm);
-static int32_t read_cfg(struct acrn_vpci *vpci, union pci_bdf bdf, uint32_t offset, uint32_t bytes, uint32_t *val);
-static int32_t write_cfg(struct acrn_vpci *vpci, union pci_bdf bdf, uint32_t offset, uint32_t bytes, uint32_t val);
+static int32_t vpci_read(struct acrn_vpci *vpci, union pci_bdf bdf, uint32_t offset, uint32_t bytes, uint32_t *val);
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c]: vpci_read_cfg

File: hypervisor/dm/vpci/vpci.c [hunk 2]
```diff
@@ -39,15 +39,15 @@
...
+static int32_t write_vcfg(struct acrn_vpci *vpci, union pci_bdf bdf, uint32_t offset, uint32_t bytes, uint32_t val);
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c]: ditto

File: hypervisor/dm/vpci/vpci_bridge.c [hunk 3]
```diff
@@ -106,7 +106,7 @@ static int32_t vpci_bridge_read_cfg(const struct pci_vdev *vdev, uint32_t offset
 	return 0;
 }
 
-static int32_t vpci_bridge_write_cfg(__unused struct pci_vdev *vdev, __unused uint32_t offset,
+static int32_t write_bridge_cfg(__unused struct pci_vdev *vdev, __unused uint32_t offset,
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci_bridge.c]: write_vpci_bridge_cfg

File: hypervisor/dm/vpci/vpci_bridge.c [hunk 4]
```diff
@@ -93,7 +93,7 @@ static void deinit_vpci_bridge(__unused struct pci_vdev *vdev)
 {
 }
 
-static int32_t vpci_bridge_read_cfg(const struct pci_vdev *vdev, uint32_t offset,
+static int32_t read_bridge_cfg(const struct pci_vdev *vdev, uint32_t offset,
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci_bridge.c]: read_vpci_bridge_cfg

File: hypervisor/dm/vpci/vpci.c [hunk 5]
```diff
@@ -179,7 +179,7 @@ static bool pci_cfgdata_io_write(struct acrn_vcpu *vcpu, uint16_t addr, size_t b
  * @retval 0 on success.
  * @retval other on false. (ACRN will deliver this MMIO request to DM to handle for post-launched VM)
  */
-static int32_t vpci_handle_mmconfig_access(struct io_request *io_req, void *private_data)
+static int32_t vpci_mmio_cfg_readwrite(struct io_request *io_req, void *private_data)
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c]: After double check with Jason. We will keep access here.

File: hypervisor/dm/vpci/vmsi.c [hunk 6]
```diff
@@ -69,12 +69,12 @@ static void remap_vmsi(const struct pci_vdev *vdev)
 	uint32_t vmsi_msgdata, vmsi_addrlo, vmsi_addrhi = 0U;
 
 	/* Read the MSI capability structure from virtual device */
-	vmsi_addrlo = pci_vdev_read_cfg_u32(vdev, capoff + PCIR_MSI_ADDR);
+	vmsi_addrlo = vdev->cfgdata.data_32[(capoff + PCIR_MSI_ADDR) >> 2U];
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vmsi.c]: In here, should we use pci_vdev_read_cfg?

File: hypervisor/dm/vpci/vmsi.c [hunk 7]
```diff
@@ -69,12 +69,12 @@ static void remap_vmsi(const struct pci_vdev *vdev)
...
 	if (vdev->msi.is_64bit) {
-		vmsi_addrhi = pci_vdev_read_cfg_u32(vdev, capoff + PCIR_MSI_ADDR_HIGH);
-		vmsi_msgdata = pci_vdev_read_cfg_u16(vdev, capoff + PCIR_MSI_DATA_64BIT);
+		vmsi_addrhi = vdev->cfgdata.data_32[(capoff + PCIR_MSI_ADDR_HIGH) >> 2U];
+		vmsi_msgdata = vdev->cfgdata.data_16[(capoff + PCIR_MSI_DATA_64BIT) >> 1U];
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vmsi.c]: ditto

File: hypervisor/dm/vpci/vmsi.c [hunk 8]
```diff
@@ -69,12 +69,12 @@ static void remap_vmsi(const struct pci_vdev *vdev)
...
 	} else {
-		vmsi_msgdata = pci_vdev_read_cfg_u16(vdev, capoff + PCIR_MSI_DATA);
+		vmsi_msgdata = vdev->cfgdata.data_16[(capoff + PCIR_MSI_DATA) >> 1U];
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vmsi.c]: ditto

File: hypervisor/dm/vpci/vmsi.c [hunk 9]
```diff
@@ -69,12 +69,12 @@ static void remap_vmsi(const struct pci_vdev *vdev)
 	uint32_t vmsi_msgdata, vmsi_addrlo, vmsi_addrhi = 0U;
 
 	/* Read the MSI capability structure from virtual device */
-	vmsi_addrlo = pci_vdev_read_cfg_u32(vdev, capoff + PCIR_MSI_ADDR);
+	vmsi_addrlo = pci_vdev_read_cfg(vdev, (capoff + PCIR_MSI_ADDR), 4U);
 	if (vdev->msi.is_64bit) {
-		vmsi_addrhi = pci_vdev_read_cfg_u32(vdev, capoff + PCIR_MSI_ADDR_HIGH);
-		vmsi_msgdata = pci_vdev_read_cfg_u16(vdev, capoff + PCIR_MSI_DATA_64BIT);
+		vmsi_addrhi = pci_vdev_read_cfg(vdev, (capoff + PCIR_MSI_ADDR_HIGH), 4U);
+		vmsi_msgdata = (uint16_t)pci_vdev_read_cfg(vdev, (capoff + PCIR_MSI_DATA_64BIT), 2U);
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vmsi.c]: no need of (uint16_t)?

File: hypervisor/dm/vpci/vmsi.c [hunk 10]
```diff
@@ -69,12 +69,12 @@ static void remap_vmsi(const struct pci_vdev *vdev)
...
 	} else {
-		vmsi_msgdata = pci_vdev_read_cfg_u16(vdev, capoff + PCIR_MSI_DATA);
+		vmsi_msgdata = (uint16_t)pci_vdev_read_cfg(vdev, (capoff + PCIR_MSI_DATA), 2U);
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vmsi.c]: ditto

**Comment by ClaudZhang1995** [comment]
OK to verify

**Comment by HuiHuangShi** [comment]
64 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/dm/vpci/vhostbridge.c func:read_vhostbridge_cfg offset:102<br> reason:Static procedure is not explicitly called in code analysed. : read_vhostbridge_cfg<br><2> config:common filename:/hypervisor/dm/vpci/vhostbridge.c func:write_vhostbridge_cfg offset:115<br> reason:Static procedure is not explicitly called in code analysed. : write_vhostbridge_cfg<br><3> config:common filename:/hypervisor/dm/vpci/vpci_bridge.c func:read_vpci_bridge_cfg offset:96<br> reason:Static procedure is not explicitly called in code analysed. : read_vpci_bridge_cfg<br><4> config:common filename:/hypervisor/dm/vpci/vpci_bridge.c func:write_vpci_bridge_cfg offset:109<br> reason:Static procedure is not explicitly called in code analysed. : write_vpci_bridge_cfg<br><5> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_read offset:50<br> reason:Pointer param should be declared pointer to const. : vcpu<br><6> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_read offset:50<br> reason:Pointer param should be declared pointer to const. : vcpu<br><7> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_read offset:50<br> reason:Static procedure is not explicitly called in code analysed. : vpci_pio_cfgaddr_read<br><8> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_write offset:73<br> reason:Pointer param should be declared pointer to const. : vcpu<br><9> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_write offset:73<br> reason:Pointer param should be declared pointer to const. : vcpu<br><10> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_write offset:73<br> reason:Static procedure is not explicitly called in code analysed. : vpci_pio_cfgaddr_write<br><11> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_write offset:85<br> reason:Implicit conversion of underlying type (MR). :  (unsigned short and unsigned int): vbdf.value = cfg_addr -> bits.bdf<br><12> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_write offset:85<br> reason:Use of mixed mode arithmetic. :  (unsigned short and unsigned int): vbdf.value = cfg_addr -> bits.bdf<br><13> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:123<br> reason:Pointer param should be declared pointer to const. : vcpu<br><14> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:123<br> reason:Pointer param should be declared pointer to const. : vcpu<br><15> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:123<br> reason:Static procedure is not explicitly called in code analysed. : vpci_pio_cfgdata_read<br><16> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.bits.resv<br><17> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.bits.reg_num<br><18> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.value<br><19> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.bits.resv<br><20> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.bits.reg_num<br><21> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.bits.bdf<br><22> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.bits.bdf<br><23> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.value<br><24> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Implicit conversion: actual to formal param (MR). :  (int and unsigned int):  param  1<br><25> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:137<br> reason:Implicit conversion of underlying type (MR). :  (unsigned short and unsigned int): bdf.value = cfg_addr.bits.bdf<br><26> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:137<br> reason:Narrower int conversion without cast. :  (unsigned short and unsigned int): bdf.value = cfg_addr.bits.bdf<br><27> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:156<br> reason:Pointer param should be declared pointer to const. : vcpu<br><28> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:156<br> reason:Pointer param should be declared pointer to const. : vcpu<br><29> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:156<br> reason:Static procedure is not explicitly called in code analysed. : vpci_pio_cfgdata_write<br><30> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:162<br> reason:Procedure contains UR data flow anomalies. : bdf.bits.d<br><31> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:162<br> reason:Procedure contains UR data flow anomalies. : bdf.bits.f<br><32> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:162<br> reason:Procedure contains UR data flow anomalies. : bdf.bits.b<br><33> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:162<br> reason:Procedure contains UR data flow anomalies. : bdf.bits.d<br><34> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:162<br> reason:Procedure contains UR data flow anomalies. : bdf.bits.b<br>.....<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
64 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/dm/vpci/vhostbridge.c func:read_vhostbridge_cfg offset:102<br> reason:Static procedure is not explicitly called in code analysed. : read_vhostbridge_cfg<br><2> config:common filename:/hypervisor/dm/vpci/vhostbridge.c func:write_vhostbridge_cfg offset:115<br> reason:Static procedure is not explicitly called in code analysed. : write_vhostbridge_cfg<br><3> config:common filename:/hypervisor/dm/vpci/vpci_bridge.c func:read_vpci_bridge_cfg offset:96<br> reason:Static procedure is not explicitly called in code analysed. : read_vpci_bridge_cfg<br><4> config:common filename:/hypervisor/dm/vpci/vpci_bridge.c func:write_vpci_bridge_cfg offset:109<br> reason:Static procedure is not explicitly called in code analysed. : write_vpci_bridge_cfg<br><5> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_read offset:50<br> reason:Pointer param should be declared pointer to const. : vcpu<br><6> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_read offset:50<br> reason:Pointer param should be declared pointer to const. : vcpu<br><7> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_read offset:50<br> reason:Static procedure is not explicitly called in code analysed. : vpci_pio_cfgaddr_read<br><8> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_write offset:73<br> reason:Pointer param should be declared pointer to const. : vcpu<br><9> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_write offset:73<br> reason:Pointer param should be declared pointer to const. : vcpu<br><10> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_write offset:73<br> reason:Static procedure is not explicitly called in code analysed. : vpci_pio_cfgaddr_write<br><11> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_write offset:85<br> reason:Implicit conversion of underlying type (MR). :  (unsigned short and unsigned int): vbdf.value = cfg_addr -> bits.bdf<br><12> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgaddr_write offset:85<br> reason:Use of mixed mode arithmetic. :  (unsigned short and unsigned int): vbdf.value = cfg_addr -> bits.bdf<br><13> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:123<br> reason:Pointer param should be declared pointer to const. : vcpu<br><14> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:123<br> reason:Pointer param should be declared pointer to const. : vcpu<br><15> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:123<br> reason:Static procedure is not explicitly called in code analysed. : vpci_pio_cfgdata_read<br><16> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.bits.resv<br><17> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.bits.reg_num<br><18> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.value<br><19> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.bits.resv<br><20> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.bits.reg_num<br><21> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.bits.bdf<br><22> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.bits.bdf<br><23> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Function return value potentially unused. : cfg_addr.value<br><24> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:134<br> reason:Implicit conversion: actual to formal param (MR). :  (int and unsigned int):  param  1<br><25> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:137<br> reason:Implicit conversion of underlying type (MR). :  (unsigned short and unsigned int): bdf.value = cfg_addr.bits.bdf<br><26> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_read offset:137<br> reason:Narrower int conversion without cast. :  (unsigned short and unsigned int): bdf.value = cfg_addr.bits.bdf<br><27> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:156<br> reason:Pointer param should be declared pointer to const. : vcpu<br><28> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:156<br> reason:Pointer param should be declared pointer to const. : vcpu<br><29> config:common filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:156<br> reason:Static procedure is not explicitly called in code analysed. : vpci_pio_cfgdata_write<br><30> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:162<br> reason:Procedure contains UR data flow anomalies. : bdf.bits.d<br><31> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:162<br> reason:Procedure contains UR data flow anomalies. : bdf.bits.f<br><32> config:sharing_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:162<br> reason:Procedure contains UR data flow anomalies. : bdf.bits.b<br><33> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:162<br> reason:Procedure contains UR data flow anomalies. : bdf.bits.d<br><34> config:part_config filename:/hypervisor/dm/vpci/vpci.c func:vpci_pio_cfgdata_write offset:162<br> reason:Procedure contains UR data flow anomalies. : bdf.bits.b<br>.....<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by ClaudZhang1995** [comment]
OK to verify

**Pull request merged**